trio/_core/_tests/test_ki.py

```diff
@@ -465,19 +466,19 @@ async def main():
             with pytest.raises(_core.Cancelled):
                 await _core.checkpoint()
 
-    _core.run(main)
+    _core.run(main_8)
```

Comment:
TBH we should experiment with the mypy option that allows for reassignment...

Reply:
The reassignment option can open the door to lots of issues though. I will say it is convenient in a lot of cases to simply reassign something, and it can become a bit of type gymnastics with especially complex code, but I think that in the end leaving it off might be the best option. I feel like a lot of subtle issues might get suppressed with the reassignment option enabled.